docs(xmllib): create overview of xmllib (DEV-5291) (#1930)
dasch-swiss/dsp-tools: docs/xmllib-docs/overview.md

We created the following function to clean your input.

```python
input_string = "This should be\na list\nof strings."
cleaned_list = xmllib.create_list_from_input(
    input_value=input_string,
    separator="\n"
)
```

You can find this and other functions to help you process your data [here](general-functions.md).

Please contact us with a feature request if you cannot find a function in the list, 
but you feel that it may have generic applications and would be helpful to other users.

### Adding A File To a Resource

In the ontology you can 
[specify](../data-model/json-project/ontologies.md#resource-super) that a resource should have a file attached.

You can add all types of files to the resource with the following function.
Please note that only one file or IIIF-URI (see below) are allowed per resource.

```python
image_resource = image_resource.add_file(
    filename="cat.tiff",
    license=xmllib.LicenseRecommended.CC.BY,
    copyright_holder="Mouse University",
    authorship=["Minnie Mouse"],
)
```

We require that legal information be provided for resources with files. 

> While the copyright holder and authorships are free text fields.
> All licenses referenced must be predefined, [look here for our recommended options](./licenses/recommended.md).
> 
> Please contact us if you cannot find the required license in either our recommended or [other options](./licenses/other.md).

Reviewer: ```suggestion
While the copyright holder and authorships are free text fields,
the license must be chosen from a predefined set.
The recommended licenses are listed [here](./licenses/recommended.md),
and some others are listed [here](./licenses/other.md).

Please contact us if you required license is not available.
```

### Adding IIIF-URIs To a Resource

We provide the option to add a IIIF-URI to a Resource with the super-class `StillImageRepresentation`.

> This way you can reference images that lie on other servers.

Reviewer: ```suggestion
This way you can reference images that are hosted on other servers.
```

```python
resource = resource.add_iiif_uri(
    iiif_uri="https://iiif.wellcomecollection.org/image/b20432033_B0008608.JP2/full/1338%2C/0/default.jpg",
    license=xmllib.LicenseRecommended.CC.BY_NC,
    copyright_holder="Wellcome Collection",
    authorship=["Cavanagh, Annie"]
)
```

Please note that the IIIF-URI must follow the official syntax specified [here](https://iiif.io/api/image/3.0/#2-uri-syntax).

### Adding a Comment To a Value

It is possible to add a comment to individual values or files with the following parameter.

```python

> resource = resource.add_integer_optional(

Reviewer: ```suggestion
resource = resource.add_integer(
```

    prop_name=":hasInt",
    value=1,
    comment="This text is a comment on this integer.",
)
```

> Please note, that if you add a comment to values with the add multiple functions, 
> all the values will have the same comment.

Reviewer: ```suggestion
Please note that if you add a comment to values with the `add_xyz_multiple()` methods, 
all values will have the same comment.
```

Reviewer: This is already quite specific, so I think this remark should be removed altogether.

### Specifying Permissions

We recommend to specify permissions on the [project level](../data-model/json-project/overview.md#default_permissions)
or on [individual classes and properties](../data-model/json-project/overview.md#default_permissions_overrule).

However, you can over-rule the default permissions with the following designated parameter.

```python
image_resource = image_resource.add_simpletext(
    prop_name=":hasRemark",
    value="Text only visible for the project.",
    permissions=xmllib.Permissions.PRIVATE,
)
```

[See the documentation for details.](permissions.md)

### Adding a Resource To the Root

> Once you added all the information to the resource you can add it to the root.

Reviewer: ```suggestion
Once you have added all information to the resource, you can add the resource to the root.
```

```python
root = root.add_resource(resource)
```

Similarly to the values, we also provide the possibility to add multiple or optional resources to the root.

### Creating DSP-Base Resources

DSP has four additional resource types that can be used out of the box without defining them in the data model.

- [`AudioSegmentResource`](./dsp-base-resources/audio-segment-resource.md) and 
  [`VideoSegmentResource`](./dsp-base-resources/video-segment-resource.md) 
  are resources that can be used to annotate a segment of an audio or video file of another resource.
- [`RegionResource`](./dsp-base-resources/region-resource.md) describes a region of a resource with an image file.
- [`LinkResource`](./dsp-base-resources/link-resource.md) is a collection of links to other resources.

These four types have pre-defined properties, 
which either are added directly when the resource is created or can be added afterwards with a dedicated function.

```python
region = xmllib.RegionResource.create_new(
    res_id="region_1",
    label="Region label",
    region_of="image_resource"
)
region = region.add_rectangle(
    point1=(0.1, 0.1),
    point2=(0.2, 0.2),
)
```

> Please consult the individual documentations, linked above for details.

Reviewer: ```suggestion
Please consult the individual documentations (linked above) for details.
```

### Writing the File

> Once you have added all the data, you can write the XML file with the following function.

Reviewer: ```suggestion
Once you have added all data, you can write the XML file with the following function.
```
